cross-modal-search/README.md

```diff
@@ -32,115 +33,153 @@
-
-Modality is an attribute assigned to a document in Jina in the protobuf Document structure.
-It is possible that documents may be of the same mime type,
-but come from different distributions,
-for them to have different modalities.
-**Example**: In an article or web page, 
-the body text and the title are from the same mime type (text),
-but can be considered of different modalities (distributions).
+![](https://github.com/jina-ai/examples/blob/master/cross-modal-search/visualizations/cross-modal-result.jpg)
 
-Different encoders map different modalities to a common embedding space.
-They need to extract semantic information from the documents. 
+## Overview
+|  |  |
+| ------------- | ------------- |
+| Learnings | How to use image captions (short descriptions of the image content) to search for images. |
+| Used for indexing | Images + captions. |
+| Used for querying | Image caption e.g. "A boy playing basketball". |
+| Dataset used | [Flickr 8k](https://www.kaggle.com/adityajn105/flickr8k) containing 8k image caption pairs. |
+| Model used | Both [CLIP: Contrastive Language-Image Pre-Training](https://arxiv.org/abs/2007.13135) and [VSE++: Improving Visual-Semantic Embeddings with Hard Negatives](https://arxiv.org/pdf/1707.05612.pdf) are demonstrated. |
 
-In this embedding space,
-documents that are semantically relevant to each other from different modalities are expected to be close to another -  Metric Learning
+In this example, Jina is used to implement a cross-modal search system.
+This example allows the user to search for images given a caption description.
+First, we encode all images in our dataset into vectors and create an index of these vectors.
+When we search with text inputs, we compute the vector representation of this text and compare it to the previously calculated image vector index to find the most similar image. 
+It is also possible to do it the other way around and use the image encoding to search for similar text-embeddings (captions).
 
-In the example, we expect images embeddings to be nearby their captions’ embeddings.
+_____
 
-**Research for Cross Modal Retrieval**
+## 🐍 Build the app with Python
 
-The models used for the example are cited from the paper, you can try our example with one of them:
+These instructions explain how to build the example yourself and deploy it with Python. 
+If you want to skip the building steps and just run the example with Docker, check out the  [Docker section](#-deploy-the-prebuild-application-using-docker) below. 
 
-1. [CLIP: Contrastive Language-Image Pre-Training](https://arxiv.org/abs/2007.13135) (recommend)
-2. [VSE++: Improving Visual-Semantic Embeddings with Hard Negatives](https://arxiv.org/pdf/1707.05612.pdf).
 
-Both of the models have been trained to encode pairs of `text` and `images` into a common embedding space.
+### 🗝️ Requirements
 
-**CLIP Encoders in Jina for Cross Modal Search**
+1. You have a working Python 3.7 or 3.8 environment. 
```

Comment:
Do we also not need Docker installed and working here? (As we found out in my pair programming)

Reply:
Just added it in https://github.com/jina-ai/examples/pull/614/commits/28376deec036581f2b8aa919c85aced1dc937979

I also checked the example again with Florian yesterday.